src/backend/dx11/src/lib.rs

```diff
@@ -709,7 +711,10 @@ pub struct CommandBuffer {
     #[derivative(Debug="ignore")]
     context: ComPtr<d3d11::ID3D11DeviceContext>,
     #[derivative(Debug="ignore")]
-    list: Option<ComPtr<d3d11::ID3D11CommandList>>
+    list: Option<ComPtr<d3d11::ID3D11CommandList>>,
+    vertex_buffers: [*mut d3d11::ID3D11Buffer; 16],
```

Comment:
similarly, is there any reason to not have a single array of structures here (as opposed to structure of arrays)?

Reply:
Compare
```rust
self.context.IASetVertexBuffers(
    0,
    d3d11::D3D11_IA_VERTEX_INPUT_RESOURCE_SLOT_COUNT,
    self.vertex_buffers.as_ptr(),
    self.vertex_strides.as_ptr(),
    self.vertex_offsets.as_ptr(),
);
```
to
```rust
for (i, (buffer, stride, offset)) in self.vertex_buffer_aos.iter().enumerate() {
    self.context.IASetVertexBuffers(
        i,
        1,
        buffer,
        stride,
        offset,
    );
}
```